components/Auction-Schedule.jsx

```diff
@@ -0,0 +1,260 @@
+		}
+	}
+
+	await Promise.all(promises)
+		.then((stamps) => {
+			for(let i = 0; i < promises.length; i++) {
+				const date = new Date(stamps[i].toPrimitive());
+				auctions[index][keys[i]] = date.toDateString();
+			}
+		})
+		.catch((error) => {
+			console.log(error);
+		})
+
+	Render(chain, auctions, setAuctions, index);
+}
+
+// Estimate a future blocks date based on 6 second block times
+function EstimateBlockDate(date, currentBlock, estimatedBlock) {
+	const blockDifference = estimatedBlock - currentBlock;
+	const seconds = blockDifference * 6;
+	const dateCopy = new Date(date.valueOf())
+	dateCopy.setSeconds(dateCopy.getSeconds() + seconds);
+	return dateCopy.toDateString();
+}
+
+// Update JSX
+function Render(chain, auctions, setAuctions, index) {
+	let explorerUrl = undefined;
+	if (chain === "Polkadot") {
+		explorerUrl = "https://polkadot.subscan.io/block/";
+	} else if (chain === "Kusama") {
+		explorerUrl = "https://kusama.subscan.io/block/";
+	}
+
+	// Current block information
+	let currentBlockNumber = ChainState.blockNumber;
+	let currentBlockDate = ChainState.blockDate;
+	if (currentBlockNumber !== undefined) {
+		currentBlockDate = currentBlockDate.toDateString();
+	} else {
+		currentBlockNumber = 0;
+		currentBlockDate = "Connecting...";
+	}
+
+	// If still calculating date estimation, inform user
+	if (auctions[index].hasOwnProperty("startDate") === false) {
+		const msg = "Retrieving date...";
+		auctions[index]["startDate"] = msg;
+		auctions[index]["endPeriodDate"] = msg;
+		auctions[index]["biddingEndsDate"] = msg;
+		auctions[index]["onboardStartDate"] = msg;
+		auctions[index]["onboardEndDate"] = msg;
+	}
+	
+	// On-boarding range
+	let onboarding = <div>
+		{`${auctions[index].onboardStartDate} - `}
+		<a href={`${explorerUrl}${auctions[index].onboardStartBlock}`}>
+			Block #{auctions[index].onboardStartBlock}
+		</a>
+		{` to `}
+		{`${auctions[index].onboardEndDate} - `}
+		<a href={`${explorerUrl}${auctions[index].onboardEndBlock}`}>
+			Block #{auctions[index].onboardEndBlock}
+		</a>
+	</div>
+	// If onboarding is too far in the future to calculate
+	if (auctions[index]["onboardStartBlock"] === 0 || auctions[index]["onboardEndBlock"] === 0) {
+		onboarding = <div>
+			On-boarding cannot yet be determined for this future event.
+		</div>
+	}
+
+	const content = <div>
+		<select id="AuctionSelector" onChange={(e) => LoadBlockCacheThenUpdate(chain, auctions, setAuctions, e)} style={{ border: '2px solid #e6007a', height: '40px' }}>
+			{Options.map((option) => (option))}
+		</select>
+		<hr />
+		<b>Current Chain State:</b>
+		<br />
+		{`${currentBlockDate} - `}
+		<a href={`${explorerUrl}${currentBlockNumber}`}>
+			Block #{currentBlockNumber}
+		</a>
+		<hr />
+		<b>Auction Starts:</b>
+		<br />
+		{`${auctions[index].startDate} - `}
+		<a href={`${explorerUrl}${auctions[index].startBlock}`}>
+			Block #{auctions[index].startBlock}
+		</a>
+		<hr />
+		<b>Auction Ends:</b>
```

Comment:
This is when bidding starts. It might be misleading to say Auction Ends. This is actually the end of the auction opening period when the bidding starts. https://wiki.polkadot.network/docs/learn-auction#mechanics-of-a-candle-auction

```suggestion
		<b>Bidding Starts:</b>
```